src/mono/mono/metadata/class-init.c

```diff
@@ -564,6 +564,11 @@ mono_class_create_from_typedef (MonoImage *image, guint32 type_token, MonoError
 		}
 	}
 
+	if (MONO_CLASS_IS_IMPORT (klass)) {
+		mono_class_set_failure_and_error (klass, error, "Built-in COM interop is not supported on the Mono runtime.");
+		goto parent_failure;
+	}
```

Comment:
Apparently this is breaking some dependency injection tests. I guess we have to be lazier - maybe we need  to keep `m_class_is_com_object` and just throw a TLE if we try to instantiate it.